pkg/cmd/infra/router/router.go

```diff
@@ -37,35 +54,75 @@ type templateRouterConfig struct {
 	StatsUsername      string
 }
 
+// f5RouterConfig is the config necessary to start an F5 router plugin
+type f5RouterConfig struct {
+	Host     string
+	Username string
+	Password string
+	Vserver  string
+	Insecure bool
+}
+
+const (
+	//routerTypeF5 is the type value for an f5 router plugin
+	routerTypeF5 = "f5"
+	//routerTypeTemplate is the type value for a template router plugin
+	routerTypeTemplate = "template"
+)
+
+func newEmptyConfig() *routerConfig {
+	return &routerConfig{
+		Config:               clientcmd.NewConfig(),
+		TemplateRouterConfig: &templateRouterConfig{},
+		F5RouterConfig:       &f5RouterConfig{},
+	}
+}
+
 // NewCommndTemplateRouter provides CLI handler for the template router backend
 func NewCommandTemplateRouter(name string) *cobra.Command {
-	cfg := &templateRouterConfig{
-		Config: clientcmd.NewConfig(),
-	}
+	cfg := newEmptyConfig()
 
 	cmd := &cobra.Command{
 		Use:   fmt.Sprintf("%s%s", name, clientcmd.ConfigSyntax),
 		Short: "Start an OpenShift router",
 		Long:  routerLong,
 		Run: func(c *cobra.Command, args []string) {
+			if !isValidType(cfg.Type) {
+				glog.Fatal("Invalid type specified.  Valid values are template, f5")
```

Comment:
change to `Fatalf` and output the type that was passed and use your constant values for the valid types